rtl/ecc_wrap/ecc_sram_wrapper.hjson

```diff
@@ -0,0 +1,80 @@
+{
+  name: "ECC_manager",
+  clock_primary: "clk_i",
+  reset_primary: "rst_ni",
+  bus_interfaces: [
+    { protocol: "reg_iface",
+      direction: "device"
+    }
+  ],
+
+  regwidth: "32",
+  registers: [
+    { name: "PRIVATE0",
+      desc: "errors in private bank 0",
+      swaccess: "rw0c",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0",
+          name: "private0",
+          desc: "Correctable mismatches caught by ecc"
+        }
+      ]
+    },
+    { name: "PRIVATE1",
+      desc: "errors in private bank 1",
+      swaccess: "rw0c",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0",
+          name: "private1",
+          desc: "Correctable mismatches caught by ecc"
+        }
+      ]
+    },
+    { name: "CUTS0",
+      desc: "errors in public bank 0",
+      swaccess: "rw0c",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0",
+          name: "cuts0",
+          desc: "Correctable mismatches caught by ecc"
+        }
+      ]
+    },
+    { name: "CUTS1",
+      desc: "errors in public bank 1",
+      swaccess: "rw0c",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0",
+          name: "cuts1",
+          desc: "Correctable mismatches caught by ecc"
+        }
+      ]
+    },
+    { name: "CUTS2",
+      desc: "errors in public bank 2",
+      swaccess: "rw0c",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0",
+          name: "cuts2",
+          desc: "Correctable mismatches caught by ecc"
+        }
+      ]
+    },
+    { name: "CUTS3",
+      desc: "errors in public bank 3",
+      swaccess: "rw0c",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0",
+          name: "cuts3",
+          desc: "Correctable mismatches caught by ecc"
+        }
+      ]
+    }
+  ],
```

Comment:
You may want to consider something more re-usable, as illustrated below. This will still require you to re-generate the registers when using a different number of banks, but can greatly improve code-reuse, instead of locking in the exact configuration in PULPissimo.

```suggestion
  param_list: [ 
    { name: "NumBanks",
      default: "6"
    }
  ],

  registers: [
    { multireg: 
      { count: "NumBanks",
        cname: "mismatch_count",
        name: "mismatch_count",
        desc: "Correctable mismatches caught by ecc",
        swaccess: "rw0c",
        hwaccess: "hrw",
        resval: "0",
        fields: [
          { bits: "31:0",
            name: "correctable_mismatches",
            desc: "Correctable mismatches caught by ecc"
          }
        ]
      }
    }
  ],
```